tests/Unit/Ads/AdsServiceTest.php

```diff
@@ -0,0 +1,73 @@
+<?php
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\GoogleListingsAndAds\Tests\Unit\Ads;
+
+use Automattic\WooCommerce\GoogleListingsAndAds\Ads\AdsService;
+use Automattic\WooCommerce\GoogleListingsAndAds\Tests\Framework\UnitTest;
+use Automattic\WooCommerce\GoogleListingsAndAds\Options\AdsAccountState;
+use Automattic\WooCommerce\GoogleListingsAndAds\Options\OptionsInterface;
+
+/**
+ * Class AdsServiceTest
+ *
+ * @package Automattic\WooCommerce\GoogleListingsAndAds\Tests\Unit\Ads
+ *
+ * @property MockObject|AdsAccountState     $state
+ * @property AdsService                     $ads_service
+ * @property OptionsInterface               $options
+ */
```

Comment:
Thanks for adding tests ❤️ 

💅  One thing I normally do and is useful for me is to add

`@group AdsService`

That way we could run the tests on this file (without writing the entire path) and other files in same group by using

`../phpunit --group AdsService`

Reply:
💅 For A++ would be cool to have test as well for the 2 created notes.

Reply:
Thanks for sharing this! I did not know about @group! I have added: [b05eae9](https://github.com/woocommerce/google-listings-and-ads/pull/1508/commits/b05eae987d482f4dd20afb3b1f231dc3e055f6e5)

>would be cool to have test as well for the 2 created notes.

I am not sure what I should test here, maybe `should_be_added`? But this is basically `ads_service->is_migration_completed` and if I test the actual NoteEntry then I will be testing a WC Admin component.